OrbitCore/SymbolsManager.cpp

```diff
@@ -134,16 +127,19 @@ void SymbolsManager::HandleResponse(const Message& message) {
   // Notify app of new debug symbols.
   core_app_->OnRemoteModuleDebugInfo(infos);
 
-  FinalizeTransaction();
+  // Finalize transaction.
+  mutex_.Lock();
+  std::shared_ptr<Session> session = id_sessions_[id];
+  id_sessions_.erase(id);
+  mutex_.Unlock();
+  FinalizeTransaction(session);
 }
 
-void SymbolsManager::FinalizeTransaction() {
+void SymbolsManager::FinalizeTransaction(std::shared_ptr<Session> session) {
```

Comment:
Does this have to be a shared_ptr or can it be a const ref to Session? Can session be nullptr? Same for ApplySession.

Reply:
It can be null, I switched to raw pointer and const ref in ApplySession.